doc/api/buffer.markdown

```diff
@@ -21,19 +21,105 @@ The `Buffer` class is a global within Node.js, making it unlikely that one
 
 ```js
-const buf1 = new Buffer(10);
-  // creates a buffer of length 10
-
-const buf2 = new Buffer([1,2,3]);
+const buf1 = Buffer.alloc(10);
+  // creates a zero-filled buffer of length 10
+  
+const buf2 = Buffer.allocUnsafe(10);
+  // creates an uninitialized buffer of length 10
+  // this is faster than calling Buffer.alloc() but the returned
+  // buffer instance might contain old data that needs to be
+  // overwritten
+
+const buf3 = Buffer.from([1,2,3]);
   // creates a buffer containing [01, 02, 03]
 
-const buf3 = new Buffer('test');
+const buf4 = Buffer.from('test');
   // creates a buffer containing ASCII bytes [74, 65, 73, 74]
 
-const buf4 = new Buffer('tést', 'utf8');
+const buf5 = Buffer.from('tést', 'utf8');
   // creates a buffer containing UTF8 bytes [74, c3, a9, 73, 74]
 ```
 
+## `Buffer.from()`, `Buffer.alloc()`, and `Buffer.allocUnsafe()`
+
+In prior version of Node.js, `Buffer` instances were created using the
+`new Buffer()` constructor function, which allocates and initializes the
+returned `Buffer` differently based on what arguments are provided:
+
+* Passing a number as the first argument to `Buffer()` (e.g. `new Buffer(10)`),
+  allocates a new `Buffer` object of the specified size. The memory allocated
+  for such `Buffer` instances is *not* initialized and *can contain sensitive
+  data*. Such `Buffer` objects *must* be initialized *manually* by using either
+  [`buf.fill(0)`][] or by writing to the `Buffer` completely. While this
+  behavior is *intentional* to improve performance, development experience has
+  demonstrated that a more explicit distinction is required between creating a
+  fast-but-uninitialized `Buffer` versus creating a slower-but-safer `Buffer`.
+* Passing a string, array, or `Buffer` as the first arguments copies the
+  passed object's data into the `Buffer`.
+* Passing an `ArrayBuffer` returns a `Buffer` that shares allocated memory with
+  the given `ArrayBuffer`.
+
+Because the behavior of `new Buffer()` changes significantly based on the type 
+of value passed as the first argument, applications that do not properly 
+validate the input arguments passed to `new Buffer()`, or that fail to 
+appropriately initialize newly allocated `Buffer` content, can inadvertently 
+introduce security and reliability issues into their code.
+
+To make the creation of `Buffer` objects more reliable and less error prone,
+the various forms of the `new Buffer()` constructor have been **deprecated**
+and replaced by separate `Buffer.from()`, `Buffer.alloc()`, and
+`Buffer.allocUnsafe()` methods. 
+
+*Developers should migrate all existing uses of the `new Buffer()` constructors 
+one of these new APIs.*
+
+* [`Buffer.from(array)`][buffer_from_array] returns a new `Buffer` containing 
+  a *copy* of the provided octets.
+* [`Buffer.from(arrayBuffer)`][buffer_from_arraybuffer] returns a new `Buffer` 
+  that *shares the same allocated memory as the given `ArrayBuffer`.
```

Comment:
It's similar but far less unsafe... enough so that having a separate factory method likely isn't worthwhile. If it did prove to be problematic, adding a separate `Buffer.wrap(arrayBuffer)` (or `fromShared`) would still be possible.